suite/tests/client-interface/drx-scattergather.dll.c

```diff
@@ -160,14 +131,24 @@ event_bb_app2app(void *drcontext, void *tag, instrlist_t *bb, bool for_trace,
     instr_t *instr;
     bool expanded = false;
     bool scatter_gather_present = false;
+    ptr_int_t val;
 
     for (instr = instrlist_first_app(bb); instr != NULL;
          instr = instr_get_next_app(instr)) {
         if (instr_is_gather(instr)) {
             scatter_gather_present = true;
-        }
-        if (instr_is_scatter(instr)) {
+        } else if (instr_is_scatter(instr)) {
             scatter_gather_present = true;
+        } else if (!one_time_mask_clobber_test) {
+            if (instr_is_mov_constant(instr, &val) && val == TEST_MASK_CLOBBER_MARKER) {
+                instr_t *next_instr = instr_get_next(instr);
+                if (next_instr != NULL) {
+                    if (instr_is_mov_constant(next_instr, &val) &&
+                        val == TEST_MASK_CLOBBER_MARKER) {
+                        one_time_mask_clobber_test = true;
```

Comment:
It looks like this flag is never cleared, so once it's found, all future kandnw will have ud2a added afterward: that does not seem right.

Reply:
I accepted that because it was not relevant for the test, since the first one gets hit anyway. Taking your suggestion into account, I made it only instrument the first one. This needs a second flag to allow for the translating case. Done.

Reply:
You would need to have a comment stating that we were aware of the never-cleared issue, then: otherwise it looks like a mistake and a bug.